### PR TITLE
feat: procedure callers respond to models

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

Work on #6526 

### Proposed Changes

Refactors the procedure caller blocks to rely on the backing data models.

### Reason for Changes

Using the data models makes it so that procedures are explicitly defined!

### Test Coverage

Enabled some tests, added some tests, reorganized some tests etc.

### Documentation

N/A

### Additional Information

Almost certainly best to review this commit-wise haha.

Dependent on #6706

### CODE REVIEW

File: tests/mocha/blocks/procedures_test.js [hunk 1]
```diff
@@ -394,13 +396,51 @@ suite('Procedures', function() {
...
+                .connect(rowBlock1.outputConnection);
+            callBlock.getInput('ARG1').connection
+                .connect(rowBlock2.outputConnection);
+    
+            procModel.deleteParameter(1);
+            procModel.insertParameter(param2, 0);
+    
+            chai.assert.isNotNull(
+              callBlock.getInput('ARG0'),
+              'Expected the first param input to exist');
```
Comment (NeilFraser) [inline tests/mocha/blocks/procedures_test.js:431]: Shouldn't these be indented more?

**Review by NeilFraser — APPROVED** [review]

File: tests/mocha/blocks/procedures_test.js (hunk 1 above)
Comment (BeksOmega) [reply tests/mocha/blocks/procedures_test.js:432]: Hmmm strange, I guess clang isn't running in our test directories.

File: blocks/procedures.js [hunk 2]
```diff
@@ -996,6 +991,72 @@ Extensions.register(
...
+     * @return {IProcedureModel} The procedure model associated with this
+     *     block.
+     */
+    getProcedureModel() {
+      return this.model_;
+    },
+
+    /**
+     * @param {string} name The name of the procedure model to find.
+     * @param {string[]} params The param names of the procedure model to find.
```
Comment (cpcallen) [inline blocks/procedures.js:1010]: `string[]` is not valid Closure type syntax.  Do you mean `Array<string>`?

This provoked a warning from `closure-make-deps`.

File: blocks/procedures.js [hunk 3]
```diff
@@ -996,6 +991,72 @@ Extensions.register(
...
+      if (!hasMatchingParams) return null;
+
+      return model;
+    },
+
+    /**
+     * Creates a procedure definition block with the given name and params,
+     * and returns the procedure model associated with it.
+     * @param {string} name The name of the procedure to create.
+     * @param {string[]} params The names of the parameters to create.
```
Comment (cpcallen) [inline blocks/procedures.js:1030]: Ditto.

**Review by cpcallen — COMMENTED** [review]
I have been investigating unexpected diagnostics appearing in `npm test` output.  Several were traced to two minor errors in this PR.

**Comment by BeksOmega** [comment]
Can fix!

**Pull request merged**